From 87d3713893afe1158a52aa2f1ab6b1342546083e Mon Sep 17 00:00:00 2001
From: Mario Tesi <mario.tesi@st.com>
Date: Fri, 10 Mar 2023 13:22:17 +0100
Subject: [PATCH 3/3] stm: add new iio event used for timesync logic

Signed-off-by: Mario Tesi <mario.tesi@st.com>
---
 include/uapi/linux/iio/types.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/uapi/linux/iio/types.h b/include/uapi/linux/iio/types.h
index a85021f0d..42f366c6a 100644
--- a/include/uapi/linux/iio/types.h
+++ b/include/uapi/linux/iio/types.h
@@ -123,6 +123,7 @@ enum iio_event_type {
 	IIO_EV_TYPE_GESTURE,
 	/* starting custom event type */
 	IIO_EV_TYPE_FIFO_FLUSH = 0x100,
+	IIO_EV_TYPE_TIME_SYNC,
 };
 
 enum iio_event_direction {
-- 
2.25.1

