From 01ad47a59eadd9ce20dfb8a89907e141061b8bc9 Mon Sep 17 00:00:00 2001
From: mario tesi <mario.tesi@st.com>
Date: Thu, 16 Sep 2021 15:35:14 +0200
Subject: [PATCH 2/2] STMEMS: Added iio type patch for STMEMS

Signed-off-by: mario tesi <mario.tesi@st.com>
---
 include/uapi/linux/iio/types.h | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/include/uapi/linux/iio/types.h b/include/uapi/linux/iio/types.h
index 22e5e58..d227164 100644
--- a/include/uapi/linux/iio/types.h
+++ b/include/uapi/linux/iio/types.h
@@ -40,6 +40,14 @@ enum iio_chan_type {
 	IIO_PH,
 	IIO_UVINDEX,
 	IIO_ELECTRICALCONDUCTIVITY,
+	IIO_SIGN_MOTION,
+	IIO_STEP_DETECTOR,
+	IIO_STEP_COUNTER,
+	IIO_TILT,
+	IIO_TAP,
+	IIO_TAP_TAP,
+	IIO_WRIST_TILT_GESTURE,
+	IIO_GESTURE,
 };
 
 enum iio_modifier {
@@ -89,6 +97,7 @@ enum iio_event_type {
 	IIO_EV_TYPE_THRESH_ADAPTIVE,
 	IIO_EV_TYPE_MAG_ADAPTIVE,
 	IIO_EV_TYPE_CHANGE,
+	IIO_EV_TYPE_FIFO_FLUSH,
 };
 
 enum iio_event_direction {
@@ -96,6 +105,8 @@ enum iio_event_direction {
 	IIO_EV_DIR_RISING,
 	IIO_EV_DIR_FALLING,
 	IIO_EV_DIR_NONE,
+	IIO_EV_DIR_FIFO_EMPTY,
+	IIO_EV_DIR_FIFO_DATA,
 };
 
 #endif /* _UAPI_IIO_TYPES_H_ */
-- 
2.7.4

